### PR TITLE
Migrate "User.AnalyticsActiveCount" to Sync by default

### PR DESCRIPTION
**Pull request by adzimzf**
#### Summary
Migrate `User.AnalyticsActiveCount` to sync by default, remove `.DO`

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/11396

### CODE REVIEW

File: app/analytics.go [hunk 1]
```diff
@@ -79,8 +79,6 @@ func (a *App) GetAnalytics(name string, teamId string) (model.AnalyticsRows, *mo
 			}()
 		}
 
-		dailyActiveChan := a.Srv.Store.User().AnalyticsActiveCount(DAY_MILLISECONDS)
```
Comment (jespino) [inline app/analytics.go]: Here we want to have the "explicit async" approach because we want to run the queries concurrently. So you have to create a channel, and run the function inside a gorouting, you can see an example right here with the `AnalyticsTeamCount()` or you can see an example in the example pull request linked in the issue.

File: app/diagnostics.go [hunk 2]
```diff
@@ -8,9 +8,10 @@ import (
 	"runtime"
 	"strings"
 
+	"github.com/segmentio/analytics-go"
```
Comment (jespino) [inline app/diagnostics.go]: Why this change?

File: app/diagnostics.go [hunk 3]
```diff
@@ -138,15 +136,14 @@ func (a *App) trackActivity() {
 	var incomingWebhooksCount int64
 	var outgoingWebhooksCount int64
 
-	dailyActiveChan := a.Srv.Store.User().AnalyticsActiveCount(DAY_MILLISECONDS)
```
Comment (jespino) [inline app/diagnostics.go]: Same here, we expect to use the "explicit async" approach.

**Comment by adzimzf** [comment]
@jespino I have some update, please take a look.
anyway my IDE said that line `mattermost-server/app/analytics.go:125`, variable `userChan` is always `nil`.
because :
```go
var userChan store.StoreChannel 
var userInactiveChan store.StoreChannel
if teamId == "" {
	userInactiveChan = a.Srv.Store.User().AnalyticsGetInactiveUsersCount()
} else {
	userChan := make(chan store.StoreResult, 1) // different with var userChan store.StoreChannel 
	go func() {
		count, err := a.Srv.Store.User().Count(model.UserCountOptions{TeamId: teamId})
		userChan <- store.StoreResult{Data: count, Err: err}
		close(userChan)
	}()
}
```

File: app/diagnostics.go (hunk 2 above)
Comment (adzimzf) [reply app/diagnostics.go]: it automatically changed by `goimport`, because `external` import must be in different block with `internal` import or `builtin` import

File: app/diagnostics.go (hunk 3 above)
Comment (jespino) [reply app/diagnostics.go]: Here, still expecting to migrate it to "explicit async" approach.

**Comment by jespino** [comment]
@adzimzf super good catch, we should remove the `:` from the `:=` inside the else, do you mind to include that change in your PR too?

File: app/diagnostics.go (hunk 3 above)
Comment (adzimzf) [reply app/diagnostics.go]: @jespino since currently there's no `explicit async` approach in this file.
I try make it as similar as with `app/diagnostic` approach. I just still need to print error if got it, so error will not just gone.

**Comment by adzimzf** [comment]
@jespino I have some update, 
anyway I removed `:` for `userChan` variable on else block.

File: app/diagnostics.go [hunk 4]
```diff
@@ -138,16 +137,25 @@ func (a *App) trackActivity() {
...
+			mlog.Error(err.Error())
+		}
+		activeUsersDailyCountChan <- store.StoreResult{Data: count, Err: err}
+		close(activeUsersDailyCountChan)
+	}()
+
+	activeUsersMonthlyCountChan := make(chan store.StoreResult, 1)
+	go func() {
+		count, err := a.Srv.Store.User().AnalyticsActiveCount(MONTH_MILLISECONDS)
+		if err != nil {
```
Comment (jespino) [inline app/diagnostics.go]: We are not login this errors anywhere else, I think is not something expect, I would expect or log everywhere or not log anywhere. so for now, to keep the code as was before, please, remove the `mlog.Error` calls here and in the other async call that you included it.

**Comment by adzimzf** [comment]
@jespino I have some update

**Comment by adzimzf** [comment]
@jespino I have merged master into this branch
